triage/interactive.js

```diff
@@ -0,0 +1,255 @@
+    summaryText += ` (${clustered.sumRecent} in last day)`;
+  }
+
+  summaryText += ` out of ${builds.runCount} builds from ${builds.getStartTime().toLocaleString()} to ${builds.getEndTime().toLocaleString()}.`
+
+  summary.innerText = summaryText;
+
+  if (clustered.length > 0) {
+    let graph = addElement(summary, 'div');
+    renderGraph(graph, clustered.allBuilds());
+  }
+
+  renderSubset(0, maxCount || 10);
+
+  drawVisibleGraphs();
+}
+
+// Render clusters until a cluster with the given key is found, then scroll to that cluster.
+// Bails out early if no cluster with the given key is known.
+function renderUntilFound(keyId) {
+  var el = null;
+  if (!clustered.byId[keyId]) {
+    return;
+  }
+  while ((el = document.getElementById(keyId)) === null) {
+    if (lastClusterRendered >= clustered.length)
+      return;
+    renderSubset(lastClusterRendered, 50);
+  }
+  el.scrollIntoView();
+
+  // expand the graph for the selected failure.
+  drawVisibleGraphs();
+}
+
+// When the user scrolls down, render more clusters to provide infinite scrolling.
+// This is important to make the first page load fast.
+// Also, trigger a debounced lazy graph rendering pass.
+function scrollHandler() {
+  if (!clustered) return;
+  if (lastClusterRendered < clustered.length) {
+    var top = document.getElementById('clusters');
+    if (top.getBoundingClientRect().bottom < 3 * window.innerHeight) {
+      renderSubset(lastClusterRendered, 10);
+    }
+  }
+  if (drawGraphsTimer) {
+    clearTimeout(drawGraphsTimer);
+  }
+  drawGraphsTimer = setTimeout(drawVisibleGraphs, 50);
+}
+
+var drawGraphsTimer = null;
+
+function drawVisibleGraphs() {
+  for (let el of document.querySelectorAll('div.graph')) {
+    if (el.children.length > 0) {
+      continue;  // already rendered
+    }
+    let rect = el.getBoundingClientRect();
+    if (0 <= rect.top + kGraphHeight && rect.top - kGraphHeight < window.innerHeight) {
+      renderGraph(el, clustered.buildsForClusterById(el.dataset.cluster));
+    }
+  }
+}
+
+// If someone clicks on an expandable node, expand it!
+function clickHandler(evt) {
+  var target = evt.target;
+  if (expand(target)) {
+    evt.preventDefault();
+    return false;
+  }
+}
+
+// Download a file from GCS and invoke callback with the result.
+// extracted/modified from kubernetes/test-infra/gubernator/static/build.js
+function get(uri, callback, onprogress) {
+  if (uri[0] === '/') {
+    // Matches /bucket/file/path -> [..., "bucket", "file/path"]
+    var groups = uri.match(/([^/:]+)\/(.*)/);
+    var bucket = groups[1], path = groups[2];
+    var url = 'https://www.googleapis.com/storage/v1/b/' + bucket + '/o/' +
+      encodeURIComponent(path) + '?alt=media';
+  } else {
+    var url = uri;
+  }
+  var req = new XMLHttpRequest();
+  req.open('GET', url);
+  req.onload = function(resp) {
+    callback(req);
+  };
+  req.onprogress = onprogress;
+  req.send();
+}
+
+// One-time initialization of the whole page.
+function load() {
```

Comment:
IMO it would still be useful to give feedback that "we're waiting for something from the server" even if we don't report percentages